contracts/bridging/WormholeBridgeForColony.sol

```diff
@@ -0,0 +1,145 @@
+      return 24; // Optimism
+    } else if (_evmChainId == 100) {
+      return 25; // xDai
+    } else {
+      return uint16(_evmChainId % 265669);
+    }
+  }
+
+  function supportedEvmChainId(uint256 _evmChainId) public pure returns (bool) {
+    return
+      _evmChainId == 1 ||
+      _evmChainId == 137 ||
+      _evmChainId == 80001 ||
+      _evmChainId == 10 ||
+      _evmChainId == 420 ||
+      _evmChainId == 100 ||
+      _evmChainId == 265669100 ||
+      _evmChainId == 265669101;
+  }
+
+  function setWormholeAddress(address _wormhole) public auth {
+    wormhole = IWormhole(_wormhole);
+  }
+
+  function setColonyNetworkAddress(address _colonyNetwork) public auth {
+    colonyNetwork = _colonyNetwork;
+  }
+
+  function getColonyNetworkAddress() public view returns (address) {
+    return colonyNetwork;
+  }
+
+  function setColonyBridgeAddress(uint256 evmChainId, address _colonyNetwork) public auth {
+    require(evmChainId <= type(uint128).max, "colony-bridge-chainid-too-large");
+    uint16 requestedWormholeChainId = evmChainIdToWormholeChainId(evmChainId);
+    colonyBridges[requestedWormholeChainId] = _colonyNetwork;
+  }
+
+  function getColonyBridgeAddress(uint256 evmChainId) public view returns (address) {
+    uint16 requestedWormholeChainId = evmChainIdToWormholeChainId(evmChainId);
+    return colonyBridges[requestedWormholeChainId];
+  }
+
+  function wormholeFormatAddressToEthereumAddress(
+    bytes32 _wormholeFormatAddress
+  ) public pure returns (address) {
+    return address(uint160(uint256(_wormholeFormatAddress)));
+  }
+
+  function receiveMessage(bytes memory _vaa) public {
+    (IWormhole.VM memory wormholeMessage, bool valid, string memory reason) = wormhole
+      .parseAndVerifyVM(_vaa);
+
+    // Check the vaa was valid
+    require(valid, reason);
+
+    // Check came from a known colony bridge
+    require(
+      wormholeFormatAddressToEthereumAddress(wormholeMessage.emitterAddress) ==
+        colonyBridges[wormholeMessage.emitterChainId],
+      "colony-bridge-bridged-tx-only-from-colony-bridge"
+    );
+
+    // We ignore sequence numbers - bridging out of order is okay, because we have our own way of handling that
+
+    // Do the thing
+
+    (bool success, bytes memory returndata) = address(colonyNetwork).call(wormholeMessage.payload);
+    if (!success) {
+      // Stolen shamelessly from
+      // https://ethereum.stackexchange.com/questions/83528/how-can-i-get-the-revert-reason-of-a-call-in-solidity-so-that-i-can-use-it-in-th
+      // If the _res length is less than 68, then the transaction failed silently (without a revert message)
+      if (returndata.length >= 68) {
+        assembly {
+          // Slice the sighash.
+          returndata := add(returndata, 0x04)
+        }
+        require(false, abi.decode(returndata, (string))); // All that remains is the revert string
+      }
+      require(false, "require-execute-call-reverted-with-no-error");
```

Comment:
I would explore abstracting this internal call flow into a helper function, as we duplicate this logic frequently.